### PR TITLE
Tell player why they can't wield the in progress craft

### LINKED ISSUE #38180: Can't continue crafting in-progress forged items when integrated toolset is active
# Describe the bug

EDIT: I have isolated this issue down to when the Extended Toolset is active. Having that active makes this reproducible. Updated the description to reflect that.

I started crafting a nodachi. I stopped crafting because it was taking too much time.

Afterwards, I could not continue crafting -- when I activated the in-progress nodachi, nothing happened, the game acts as if I didn't select anything.

This also happened with in-progress sheet metal in my game.

However, I could not reproduce this issue from a "from-scratch" game. I have included a save game below.

# Steps To Reproduce

(I don't have reliable repro, this is in my attached saved game.)

- Craft a nodachi.
- Stop the crafting in progress (I stopped at 25%).
- Attempt to resume by activating the in progress item in inventory.
- Nothing happens -- the menu exits with no effect.

# Expected behavior

You can resume.

# Screenshots

image

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.D-12404-g1d23fcc [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Beta National Guard Camp [national_guard_camp],
    More Locations [more_locations],
    Fuji's More Buildings [FujiStruct],
    More City Locations [cityside],
    Urban Development [Urban_Development],
    No Fungal Monsters [No_Fungi]
]

# Additional context

I could not repro this in a fresh game, spawning in items in debug mode. I wonder if it has something to do with having multiple sources of charcoal / forge / tools around? My character had an integrated toolset, so that may be involved. This is a good issue to look into for someone who likes a good mystery. I tried repro'ing this in a fresh game without luck but I suspect it is an interaction with the items I had in my base.

[Yellow Lake.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4230343/Yellow.Lake.zip)

(Affected character is Adam 'Stitch' Patton)

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Bugfixes "Tell player why they can't wield the in progress craft"```

#### Purpose of change
Fixes #38180 

#### Describe the solution
Add a message when the player fails to wield the craft

#### Testing
image

#### Additional context

image

### CODE REVIEW

**Comment by Ramza13** [comment]
If you use call can_wield() it will return you a more informative reason as to why you can't wield something.

**Comment by Fris0uman** [comment]
can_wield() doesn't seem to have a case for already wielding an item that you can't drop like the extended toolset

**Comment by Ramza13** [comment]
Well it seems like adding one and using it would be helpful and fix other interactions with the extended toolkit.  Or just leave that part and call can_wield instead of the broken arm stuff?

File: src/player.cpp [hunk 1]
```diff
@@ -3137,6 +3137,11 @@ ret_val<bool> player::can_wield( const item &it ) const
                    _( "You need at least one arm to even consider wielding something." ) );
     }
 
+    if( is_armed() && weapon.has_flag( "NO_UNWIELD" ) ) {
+        return ret_val<bool>::make_failure( _( "The %s is preventing you from wielding the %s" ),
```
Comment (Night-Pryanik) [inline src/player.cpp]: Period in the end.

**Pull request merged**